libs/extended/src/Network/AMQP/Extended.hs

```diff
@@ -65,12 +66,15 @@ openConnectionWithRetries l host port vHost hooks = do
       -- Jittered exponential backoff with 1ms as starting delay and 5s as max
       -- delay.
       let policy = capDelay 5_000_000 $ fullJitterBackoff 1000
-          logError willRetry e retryStatus = do
+          logError willRetry e retryStatus = when (isSync e) $ do
             Log.err l $
               Log.msg (Log.val "Failed to connect to RabbitMQ")
                 . Log.field "error" (displayException @SomeException e)
                 . Log.field "willRetry" willRetry
                 . Log.field "retryCount" retryStatus.rsIterNumber
+          isSync e = case E.fromException e of
+            Just (SomeAsyncException _) -> False
+            _ -> True
```

Comment:
I think I made a mistake here, we're not supposed to recover from async exceptions. Could you please use [skipAsyncExceptions](https://hackage.haskell.org/package/retry-0.9.3.1/docs/Control-Retry.html#v:skipAsyncExceptions) before the `logRetries` handler in the `recovering` call?

Reply:
Yeah, that makes sense. I was confused as to why we were retrying on async exceptions, but I was trying to keep the behaviour unchanged.